### PR TITLE
Kit receipt validation

### PR DESCRIPTION
**Pull request by amber-emmes**
Changed kit receipt logic to:
1. Run in a transaction to prevent the risk of partial data updates in error cases
2. Validate that the Collection ID matches the Collection ID (Cup ID) that was linked to the Return Kit Tracking Number during Kit Assembly

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -2400,77 +2400,95 @@ const confirmShipmentKit = async (shipmentData) => {
...
+        let toReturn;
+        await db.runTransaction(async (transaction) => {
+            const kitSnapshot = await transaction.get(db.collection("kitAssembly").where('972453354', '==', package['972453354']).where('221592017', '==', 277438316));
+            if (kitSnapshot.size === 0) {
+                toReturn = false;
+                return;
+            }
+            const kitDoc = kitSnapshot.docs[0];
+            const kitData = kitDoc.data();
+            const Connect_ID = kitData['Connect_ID'];
```
Comment (jhflorey) [inline utils/firestore.js:2412]: `Connect_ID` should be camelCase - connectId

File: utils/firestore.js [hunk 2]
```diff
@@ -2400,77 +2400,95 @@ const confirmShipmentKit = async (shipmentData) => {
...
-        const Connect_ID = kitDoc.data()['Connect_ID'];
-   
-        const participantSnapshot = await db.collection("participants").where('173836415.266600170.319972665.687158491', '==', kitDoc.data()[687158491]).get();
-        const participantDoc = participantSnapshot.docs[0];
-        const participantDocData = participantSnapshot.docs[0].data();
+        let toReturn;
+        await db.runTransaction(async (transaction) => {
+            const kitSnapshot = await transaction.get(db.collection("kitAssembly").where('972453354', '==', package['972453354']).where('221592017', '==', 277438316));
+            if (kitSnapshot.size === 0) {
+                toReturn = false;
```
Comment (jhflorey) [inline utils/firestore.js:2407]: Do we need assign value to `toReturn`?

File: utils/firestore.js [hunk 3]
```diff
@@ -2400,77 +2400,95 @@ const confirmShipmentKit = async (shipmentData) => {
...
+            const prefEmail = participantDocData['869588347'];
+            const ptName = participantDocData['153211406'] || participantDocData['399159511']
+            const surveyStatus = participantDocData['547363263']
+
+            const prevParticipantObject = participantDocData[173836415][266600170][319972665];
+            const collectionId = package['259846815']?.split(' ')[0];
+            const objectId = package['259846815']?.split(' ')[1];
+            
+            if (objectId === undefined || collectionId === undefined) {
+                toReturn = { status: 'Check Collection ID' };
```
Comment (jhflorey) [inline utils/firestore.js:2430]: Do we need assign value to `toReturn`?

File: utils/firestore.js (hunk 2 above)
Comment (amber-emmes) [reply utils/firestore.js:2407]: In the original, it was `return false` here, so I'm doing my best to mirror that exact setup.

File: utils/firestore.js (hunk 3 above)
Comment (amber-emmes) [reply utils/firestore.js:2430]: In the original, it was `return { status: 'Check Collection ID' }` here, so I'm doing my best to mirror that exact setup.

**Review by anthonypetersen — APPROVED** [review]
very nice

**Review by jhflorey — APPROVED** [review]

**Pull request merged**